ksql-common/src/test/java/io/confluent/ksql/util/timestamp/StringToTimestampParserTest.java

```diff
@@ -0,0 +1,177 @@
+package io.confluent.ksql.util.timestamp;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeMatcher;
+import org.junit.Test;
+
+public class StringToTimestampParserTest {
+
+  private static final ZoneId ZID = ZoneId.systemDefault();
+  private static final ZoneId GMT_3 = ZoneId.of("GMT+3");
+  private static final ZoneId IGNORED = ZID;
+
+  private static final ZonedDateTime EPOCH =
+      ZonedDateTime.of(1970, 1, 1, 0, 0, 0, 0, ZID);
+
+  private static final ZonedDateTime FIFTH_OF_NOVEMBER =
+      ZonedDateTime.of(1605, 11, 5, 0, 0, 0, 0, ZID);
+
+  @Test
+  public void shouldParseBasicLocalDate() {
+    // Given
+    final String format = "yyyy-MM-dd HH";
+    final String timestamp = "1605-11-05 10";
+
+    // When
+    ZonedDateTime ts = new StringToTimestampParser(format).parseZoned(timestamp, ZID);
+
+    // Then
+    assertThat(ts, sameInstant(
+        FIFTH_OF_NOVEMBER
+            .withHour(10)
+            .withZoneSameInstant(ZID)));
+  }
+
+  @Test
+  public void shouldConvertToMillis() {
+    // Given
+    final String format = "yyyy-MM-dd HH";
+    final String timestamp = "1605-11-05 10";
+
+    // When
+    long ts = new StringToTimestampParser(format).parse(timestamp);
+
+    // Then
+    assertThat(ts, is(
+        FIFTH_OF_NOVEMBER
+            .withHour(10)
+            .withZoneSameInstant(ZID)
+            .toInstant()
+            .toEpochMilli()));
+  }
+
+  @Test
+  public void shouldParseFullLocalDate() {
```

Comment:
A full date would technically be down to the millisecond right?

Reply:
I think they even support nanoseconds - I'll add all those in